featuretools/feature_discovery/LiteFeature.py

```diff
@@ -40,37 +41,134 @@ def __init__(
-        self.primitive = primitive
-        self.base_features = base_features if base_features else []
-        self.df_id = df_id
-        self.idx = idx if idx is not None else 0
-        self.related_features = related_features if related_features else set()
-
-        if self.primitive:
-            assert isinstance(self.primitive, PrimitiveBase)
+        self._name = name
+        self._logical_type = logical_type
+        self._tags = tags if tags else set()
+        self._primitive = primitive
+        self._base_features = base_features if base_features else []
+        self._df_id = df_id
+        self._idx = idx if idx is not None else 0
+        self._related_features = related_features if related_features else set()
+
+        if self._primitive:
+            assert isinstance(self._primitive, PrimitiveBase)
             assert (
                 len(self.base_features) > 0
             ), "there must be base features if given a primitive"
-            self.n_output_features = self.primitive.number_output_features
-            self.depth = max([x.depth for x in self.base_features]) + 1
-            self._gen_name = self.primitive.generate_name(
-                [x.get_name() for x in self.base_features],
+            self._n_output_features = self._primitive.number_output_features
+            self._depth = max([x.depth for x in self.base_features]) + 1
+            self._alias = self._name
+            self._name = self._primitive.generate_name(
+                [x.name for x in self.base_features],
             )
 
-        elif self.name is None:
+        elif self._name is None:
             raise TypeError("Name must be given if origin feature")
-        elif self.logical_type is None:
+        elif self._logical_type is None:
             raise TypeError("Logical Type must be given if origin feature")
-        else:
-            self._gen_name = self.name
 
-        if self.logical_type is not None and "index" not in self.tags:
-            self.tags = self.tags | self.logical_type.standard_tags
+        if self._logical_type is not None and "index" not in self._tags:
+            self._tags = self._tags | self._logical_type.standard_tags
+
+        self._id = self._generate_hash()
+
+    @property
+    def name(self):
+        if self._alias:
+            return self._alias
+        elif self.is_multioutput():
+            return f"{self._name}[{self.idx}]"
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._alias = value
+
+    @property
+    def non_indexed_name(self):
+        if not self.is_multioutput():
+            raise ValueError("only used on multioutput features")
+        return self._name
+
+    @property
+    def logical_type(self):
+        return self._logical_type
+
+    @logical_type.setter
+    def logical_type(self, _):
+        raise AttributeError("logical_type is immutable")
+
+    @property
+    def tags(self):
+        return self._tags
```

Comment:
You'll need to return a copy of `self._tags` or a user can mutate it

```python
    f1 = LiteFeature(
        name="Column 1",
        logical_type=Double,
    )

    f1.tags.add("hi")
    assert "hi" in f1.tags
```

Reply:
thank you! I think this also applies to what I just asked too ;)